api/envoy/extensions/filters/network/http_connection_manager/v3/http_connection_manager.proto

```diff
@@ -195,6 +195,27 @@ message HttpConnectionManager {
     //   Such a constraint is inherent to OpenCensus itself. It cannot be overcome without changes
     //   on OpenCensus side.
     config.trace.v3.Tracing.Http provider = 9;
+
+    // Create separate tracing span for each upstream request if true. And if this flag is set to true,
+    // the tracing provider will assume that Envoy will be independent hop in the trace chain and may
+    // set span type to client or server based on this flag.
+    // This will deprecate the
+    // :ref:`start_child_span <envoy_v3_api_field_extensions.filters.http.router.v3.Router.start_child_span>`
+    // in the router.
+    //
+    // Users should set appropriate value based on their tracing provider and actual scenario:
+    //
+    // * If Envoy is used as sidecar and users want to make the sidecar and its application as only one
+    //   hop in the trace chain, this flag should be set to false. And please also make sure the
+    //   :ref:`start_child_span <envoy_v3_api_field_extensions.filters.http.router.v3.Router.start_child_span>`
+    //   in the router is not set to true.
+    // * If Envoy is used as gateway or independent proxy, or users want to make the sidecar and its
+    //   application as different hops in the trace chain, this flag should be set to true.
+    // * If tracing provider that has explicit requirements on span creation (like SkyWalking),
+    //   this flag should be set to true.
+    //
+    // The default value is false for now for backward compatibility.
+    google.protobuf.BoolValue spawn_upstream_span = 10;
```

Comment:
I initially want to just name it `start_child_span`, like the field in the route. But `start_child_span` has no explicit meaning to tell this field related to `upstream request`, so finally I name it `spawn_upstream_span`.

Reply:
Pedantically, I don't think "spawn" and "start" really mean anything different so I actually find it more confusing that these are named differently. Perhaps they should be the same name?

Furthermore, are there every any situations in which we would want to override the tracing config in the router? It seems like this is making the available controls more coarse?

I think what we would help this PR (and users in general) would be to make sure all of this is very clearly explained in the tracing architecture overview (or wherever we document tracing best). I think your entire explanation in the PR description should be in the docs somewhere. So let's start there and then we can review further? Thank you.

/wait

Reply:
> Pedantically, I don't think "spawn" and "start" really mean anything different so I actually find it more confusing that these are named differently. Perhaps they should be the same name?

I think the difference not between "spawn" and "start" but between "upstream" and "child". Because the envoy always will create child span for async http/grpc request even the `start_child_span` is false. So, I think maybe the `child` is not enough to tell the actual meaning of this field.
Although I have no very strong view on this. So, if you insist on this point, I am OK to update it.

Reply:
> Furthermore, are there every any situations in which we would want to override the tracing config in the router? It seems like this is making the available controls more coarse?

I think the old field `start_child_span` in the router and the new `spawn_upstream_span` has same scope of action: upstream span creation of current HCM.
So, basically, they are complete same, except the new field is available for the tracing provider (this is important to make the provider work correctly.)

Reply:
> I think what we would help this PR (and users in general) would be to make sure all of this is very clearly explained in the tracing architecture overview (or wherever we document tracing best). I think your entire explanation in the PR description should be in the docs somewhere. So let's start there and then we can review further? Thank you.

Good point, will do that. Because I am not the initial designer of the tracing system, if there are anything wrong, please help me correct it.